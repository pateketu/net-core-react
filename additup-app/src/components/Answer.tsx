import * as React from 'react';
import {AnswerInput} from './AnswerInput';
import IAnswerProps from './IAnswerProps';

interface IAnswerState {
    answer: number;
}
export default class Answer extends React.PureComponent<IAnswerProps, IAnswerState> {
    constructor(props: IAnswerProps) {
        super(props);
        this.state = {answer: 0};
    }
    public render() {
        return  <div>
            <AnswerInput a={this.props.a}
                         b={this.props.b}
                         onAnswer={(answer: number) => {
                            this.setState({answer});
                        }}
            ></AnswerInput>
            {isNaN(this.state.answer)
                && <div className="invalid">Please provide a valid number </div>}
            <button disabled={isNaN(this.state.answer)}
                onClick={() => {this.props.onAnswer(this.state.answer); }}>Asnwer</button>
        </div>;
    }

}
