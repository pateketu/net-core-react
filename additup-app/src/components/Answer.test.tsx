import * as React from 'react';
import renderer from 'react-test-renderer';
import Answer from './Answer';
import {mount} from 'enzyme';

describe('Answer  Component', () => {
    it('Renders expected html mark-up', () => {
        // tslint:disable-next-line:no-empty
        const tree = renderer.create(<Answer a={1} b={2} onAnswer={() => {}}></Answer>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders an error message & disables button for an invalid answer entry', () => {
        const wrapper = mount(<Answer a={1} b={2}
            // tslint:disable-next-line:no-empty
            onAnswer={(answer: number ) => { }}></Answer>);

        wrapper.find('input').simulate('change', {target: {value: 'foo'}});

        expect(wrapper.find('.invalid')).toHaveLength(1);
        expect(wrapper.find('button').prop('disabled')).toBeTruthy();
    });

    it('Clicking on Answer button fires onAnswer with expected answer value', () => {
        let result: number = 0;
        const value: number = 20;

        const wrapper = mount(<Answer a={1} b={2}
            onAnswer={(answer: number ) => { result = answer; }}></Answer>);

        wrapper.find('input').simulate('change', {target: {value: String(value)}});
        wrapper.find('button').simulate('click');

        expect(result).toBe(value);
    });

    it('Hitting enter fires onAnswer with expected answer value', () => {
        let result: number = 0;
        const value: number = 20;

        const wrapper = mount(<Answer a={1} b={2}
            onAnswer={(answer: number ) => { result = answer; }}></Answer>);

        wrapper.find('input').simulate('change', {target: {value: String(value)}});
        wrapper.find('form').simulate('submit');

        expect(result).toBe(value);
    });

});
