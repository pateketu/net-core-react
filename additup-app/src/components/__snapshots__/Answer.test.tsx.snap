// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer  Component Renders expected html mark-up 1`] = `
<div>
  <div>
    <div
      className="equation"
    >
      <h4>
        1
      </h4>
      <h4>
        +
      </h4>
      <h4>
        2
      </h4>
      <div>
        <input
          onChange={[Function]}
          type="text"
        />
      </div>
    </div>
  </div>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Asnwer
  </button>
</div>
`;
